### PR TITLE
Handled NullPointerException in SemanticService for missing fields.

### PR DESCRIPTION
**Pull request by zkxxdiv**
### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

**Comment by zkxxdiv** [comment]
Raised Pull request for the issue https://github.com/eiffel-community/eiffel-remrem-semantics/issues/117

File: src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java [hunk 1]
```diff
@@ -203,8 +203,18 @@ public String generateMsg(String msgType, JsonObject bodyJson) {
...
-            JsonObject eventNodes = bodyJson.get(EVENT_PARAMS).getAsJsonObject();
+            JsonObject msgNodes = null;
+            JsonObject eventNodes = null;
+            if (bodyJson.get(MSG_PARAMS) != null && bodyJson.get(EVENT_PARAMS) != null
+                    && bodyJson.get(MSG_PARAMS).getAsJsonObject().get(META) != null) {
+                msgNodes = bodyJson.get(MSG_PARAMS).getAsJsonObject();
+                eventNodes = bodyJson.get(EVENT_PARAMS).getAsJsonObject();
+            } else {
+                String Message = "Found missing fields in the body JSON";
+                String Cause = "Either of eventParams or msgParams or meta in msgParams is missed";
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java]: Either is not the correct wording,

eventParams or msgParams or meta in msgParams are incorrect or missed.

File: src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java [hunk 2]
```diff
@@ -203,8 +203,18 @@ public String generateMsg(String msgType, JsonObject bodyJson) {
...
-            JsonObject msgNodes = bodyJson.get(MSG_PARAMS).getAsJsonObject();
-            JsonObject eventNodes = bodyJson.get(EVENT_PARAMS).getAsJsonObject();
+            JsonObject msgNodes = null;
+            JsonObject eventNodes = null;
+            if (bodyJson.get(MSG_PARAMS) != null && bodyJson.get(EVENT_PARAMS) != null
+                    && bodyJson.get(MSG_PARAMS).getAsJsonObject().get(META) != null) {
+                msgNodes = bodyJson.get(MSG_PARAMS).getAsJsonObject();
+                eventNodes = bodyJson.get(EVENT_PARAMS).getAsJsonObject();
+            } else {
+                String Message = "Found missing fields in the body JSON";
```
Comment (SantoshNC68) [inline src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java]: Missing fields found in body JSON.

File: src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java (hunk 1 above)
Comment (erjamni-raja) [reply src/main/java/com/ericsson/eiffel/remrem/semantics/SemanticsService.java]: _eventParams_ or _msgParams_ or _msgParams.meta_ are missed

**Pull request merged**